File: examples/simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/screens/OtherKey.java
package com.zhuinden.navigationexamplefrag.screens;

import com.google.auto.value.AutoValue;
import com.zhuinden.navigationexamplefrag.core.navigation.BaseFragment;
import com.zhuinden.navigationexamplefrag.core.navigation.BaseKey;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class OtherKey extends BaseKey {
    public static OtherKey create() {
        return new AutoValue_OtherKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new OtherFragment();
    }
}

File: simple-stack-example-kotlin/src/main/kotlin/com/zhuinden/simplestackexamplekotlin/OtherKey.kt
package com.zhuinden.simplestackexamplekotlin

import android.os.Parcelable
import paperparcel.PaperParcel

/**
 * Created by Owner on 2017.11.13.
 */
@PaperParcel
object OtherKey : BaseKey() {
    override fun createFragment(): BaseFragment = OtherFragment()

    @JvmField val CREATOR: Parcelable.Creator<OtherKey> = PaperParcelOtherKey.CREATOR

    override fun toString(): String = javaClass.name // you NEED to implement this in an `object` because of Kotlin!
}
